src/kernels/helpers.ts

```diff
@@ -2138,6 +2150,7 @@ export async function wrapKernelMethodImpl(
                     deadKernelAction: restarted ? 'deadKernelWasRestarted' : 'deadKernelWasNoRestarted'
                 };
             } else {
+                onAction(context, kernel);
```

Comment:
Fixes https://github.com/microsoft/vscode-jupyter/issues/9454

Not sure whether the name `onAction` is good, perhaps `notifyAction` might be better, not sure.